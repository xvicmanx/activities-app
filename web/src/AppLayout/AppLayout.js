import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { ToastContainer } from 'react-toastify';
import { Route, Switch, Redirect, NavLink } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faUsers, faHome, faBuilding } from '@fortawesome/free-solid-svg-icons'

import {
  NavBar,
  UserBasicInfo,
  LoadingIndicator
} from '../components';
import homeIcon from "../assets/images/front.svg";

import {
  readTokenFromCookie,
  loadUserFromToken,
} from '../entities/users/redux/UsersActions';
import './AppLayout.css';

import HomePage from '../core/pages/HomePage/HomePage';
import ActivitiesPage from '../entities/activities/pages/ActivitiesPage';
import CommunitiesPage from '../entities/communities/pages/CommunitiesPage';
import CommunityDetailsPage from '../entities/communities/pages/CommunityDetailsPage';
import UsersPage from '../entities/users/pages/UsersPage';


const AppLayout = () => {
  const { Users } = useSelector(state => state);
  const dispatch = useDispatch();
  const token = readTokenFromCookie();

  const shouldLoadToken = !Users.data && !Users.loading && token;

  useEffect(() => {
    dispatch(loadUserFromToken({
      token,
      shouldLoadToken,
    }));
  }, [dispatch, token, shouldLoadToken]);

  if (!Users.data && !Users.loading && !token) {
    return <Redirect to="/" />;
  }

  if (!Users.data) {
    return (
      <LoadingIndicator />
    );
  }

  const showMobileMenu = window.location.pathname.includes('display-menu');

  return (
    <div className="LayoutWrapper">
      <div className="Layout">
        <div className={`Layout__Left-Navigator ${showMobileMenu ? 'show' : ''}`}>
          <UserBasicInfo
            user={Users.data}
            image={Users.data.profileURL}
          />
          <div className="Options">
            <NavLink exact className="Option__Link" to="/home">
              <FontAwesomeIcon icon={faHome} className="Option__Icon" /> Inicio
            </NavLink>
            <NavLink exact className="Option__Link" to="/activities">
              <FontAwesomeIcon icon={faUsers} className="Option__Icon" /> Actividades
            </NavLink>
            <NavLink exact className="Option__Link" to="/communities">
              <FontAwesomeIcon icon={faBuilding} className="Option__Icon" /> Comunidades
            </NavLink>
            <NavLink exact className="Option__Link" to="/users">
              <FontAwesomeIcon icon={faUsers} className="Option__Icon" /> Usuarios
            </NavLink>
          </div>
        </div>
        <div className="Layout__Content">
          <NavBar right />
          <div className={`Layout__Scroll-Content-Wrapper ${showMobileMenu ? 'hide' : ''}`}>
            <div className="Layout__Scroll-Content">
              <Switch>
                <Route path="/activities" component={ActivitiesPage} />
                <Route path="/communities" component={CommunitiesPage} exact />
                <Route path="/communities/:id" component={CommunityDetailsPage} />
                <Route path="/users" component={UsersPage} />
                <Route path="/home" component={HomePage} />
              </Switch>
              <ToastContainer />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default AppLayout;
