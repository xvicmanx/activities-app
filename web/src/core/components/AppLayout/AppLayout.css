.LayoutWrapper {
  background: #919ba2;
  min-height: 100vh;
}

.Layout {
  background: #fff;
  box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23);
}

.Layout__Left-Navigator {
  background: #fff;
  display: none;
}

.Layout__Left-Navigator.show {
  display: block;
}

.Layout__Left-Navigator .Options {
  border-top: 1px solid #00000021;
  padding-top: 30px;
  padding-bottom: 10px;
}

.Layout__Left-Navigator .Options .Option__Link {
  line-height: 50px;
  padding: 0 20px;
  color: #919ba2;
  width: 100%;
  display: block;
  font-size: 1.1rem;
}

.Layout__Left-Navigator .Option__Icon {
  width: 1.5rem;
  display: inline-block;
  vertical-align: middle;
  margin-right: 10px;
}

.Layout__Left-Navigator .Options .Option__Link:hover,
.Layout__Left-Navigator .Options .Option__Link.active{
  background-color: rgba(0,0,0,0.05);
  border-left: 3px solid #5caaf6;
  color: #5caaf6;
}

.Layout__Content {
  background: #f9f9f9;
  flex-grow: 1;
  position: relative;
}

.Layout__Scroll-Content-Wrapper {
  height: calc(100vh - 115px);
  overflow: hidden;
}

.Layout__Scroll-Content {
  overflow: auto;
  max-height: calc(100vh - 75px);
  padding-bottom: 100px;
}

.Layout__Scroll-Content-Wrapper.hide {
  display: none;
}

@media only screen and (min-width: 1024px) {
  .LayoutWrapper {
    padding: 20px;
  }

  .Layout {
    display: flex;
    height: calc(100vh - 40px);
  }

  .Layout__Scroll-Content {
    max-height: calc(100vh - 95px);
  }

  .Layout__Left-Navigator {
    height: calc(100vh - 40px);
    width: 100%;
    max-width: 240px;
    min-width: 220px;;
    display: block;
    border-right: 1px solid #f3f3f3;
  }

  .Layout__Left-Navigator .Options {
    border-top: none;
    padding-top: 10px;
  }

  .Layout__Left-Navigator .Options .Option__Link {
    font-size: 1rem;
  }
}