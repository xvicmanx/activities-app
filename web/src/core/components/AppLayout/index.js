// @flow

import React from 'react';
import { NavLink } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { logOutUser } from '../../../entities/users/redux/UsersActions';

import { NavBar, UserBasicInfo } from '../';

import './AppLayout.css';

type NavigationLinkOption = {
  path: string,
  icon: any,
  label: string,
};

export type Props = {
  user: Object,
  navigationLinks: Array<NavigationLinkOption>,
  children: any,
};

const AppLayout = ({
  user,
  navigationLinks,
  children,
}: Props): React$Element<'div'> => {
  const dispatch = useDispatch();
  const showMobileMenu = window.location.pathname.includes('display-menu');
  return (
    <div className="LayoutWrapper">
      <div className="Layout">
        <div
          className={`Layout__Left-Navigator ${showMobileMenu ? 'show' : ''}`}
        >
          <UserBasicInfo user={user} />
          <div className="Options">
            {navigationLinks.map((link) => (
              <NavLink
                exact
                className="Option__Link"
                to={link.path}
                key={link.path}
              >
                <FontAwesomeIcon icon={link.icon} className="Option__Icon" />{' '}
                {link.label}
              </NavLink>
            ))}
          </div>
        </div>
        <div className="Layout__Content">
          <NavBar
            onLogoutClick={() => dispatch(logOutUser())}
          />
          <div
            className={`Layout__Scroll-Content-Wrapper ${
              showMobileMenu ? 'hide' : ''
            }`}
          >
            <div className="Layout__Scroll-Content">{children}</div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default AppLayout;
