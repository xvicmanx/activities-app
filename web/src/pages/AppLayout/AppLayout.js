import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { ToastContainer } from 'react-toastify';
import { Route, Switch, Redirect, NavLink } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faUsers, faHome, faBuilding } from '@fortawesome/free-solid-svg-icons'

import {
  HomePage,
  ActivitiesPage,
  CommunitiesPage,
  UsersPage,
} from '../';
import {
  NavBar,
  UserBasicInfo,
  LoadingIndicator
} from '../../components';
import homeIcon from "../../assets/images/front.svg";

import {
  readTokenFromCookie,
  loadUserFromToken,
} from '../../redux/Users/UsersActions';
import './AppLayout.css';

const AppLayout = () => {
  const { Users } = useSelector(state => state);
  const dispatch = useDispatch();
  const token = readTokenFromCookie();

  const shouldLoadToken = !Users.data && !Users.loading && token;

  useEffect(() => {
    dispatch(loadUserFromToken({
      token,
      shouldLoadToken,
    }));
  }, [dispatch, token, shouldLoadToken]);

  if (!Users.data && !Users.loading && !token) {
    return <Redirect to="/" />;
  }

  if (!Users.data) {
    return (
      <LoadingIndicator />
    );
  }

  const showMobileMenu = window.location.pathname.includes('display-menu');

  return (
    <div className="LayoutWrapper">
      <div className="Layout">
        <div className={`Layout__Left-Navigator ${showMobileMenu ? 'show' : ''}`}>
          <UserBasicInfo
            user={Users.data}
            image={Users.data.profileURL}
          />
          <div className="Options">
            <NavLink exact className="Option__Link" to="/home">
              <FontAwesomeIcon icon={faHome} className="Option__Icon" /> Inicio
            </NavLink>
            <NavLink exact className="Option__Link" to="/activities">
              <FontAwesomeIcon icon={faUsers} className="Option__Icon" /> Actividades
            </NavLink>
            <NavLink exact className="Option__Link" to="/communities">
              <FontAwesomeIcon icon={faBuilding} className="Option__Icon" /> Comunidades
            </NavLink>
            <NavLink exact className="Option__Link" to="/users">
              <FontAwesomeIcon icon={faUsers} className="Option__Icon" /> Usuarios
            </NavLink>
          </div>
        </div>
        <div className="Layout__Content">
          <NavBar right />
          <div className={`Layout__Scroll-Content-Wrapper ${showMobileMenu ? 'hide' : ''}`}>
            <div className="Layout__Scroll-Content">
              <Switch>
                <Route path="/activities" component={ActivitiesPage} />
                <Route path="/communities" component={CommunitiesPage} />
                <Route path="/users" component={UsersPage} />
                <Route path="/home" component={HomePage} />
              </Switch>
              <ToastContainer />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default AppLayout;
