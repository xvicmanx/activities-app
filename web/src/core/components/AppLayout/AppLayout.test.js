import React from 'react';
import renderer from 'react-test-renderer';
import { faUsers, faHome } from '@fortawesome/free-solid-svg-icons';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store'

import AppLayout from './';

const mockStore = configureMockStore()
const store = mockStore({
  Users: { data: {} },
});

describe('AppLayout', () => {
  it('renders properly', () => {
    const component = renderer.create(
      <Router>
        <Provider store={store}>
          <AppLayout
            user={{ name: 'John', profileURL: 'test.com' }}
            navigationLinks={[
              {
                path: '/test',
                label: 'Test',
                icon: faHome,
              },
              {
                path: '/another',
                label: 'Another',
                icon: faUsers,
              },
            ]}
          >
            Test
          </AppLayout>
        </Provider>
      </Router>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});