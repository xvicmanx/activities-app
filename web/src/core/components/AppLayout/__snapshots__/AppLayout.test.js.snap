// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout renders properly 1`] = `
<div
  className="LayoutWrapper"
>
  <div
    className="Layout"
  >
    <div
      className="Layout__Left-Navigator "
    >
      <div
        className="UserBasicInfo"
      >
        <figure
          className="image UserBasicInfo__Picture"
        >
          <img
            alt="Pic"
            className=""
            onError={[Function]}
            src="test.com"
          />
        </figure>
        <div>
          <h1
            className="inverted title is-4"
          >
            John
          </h1>
        </div>
      </div>
      <div
        className="Options"
      >
        <a
          aria-current={null}
          className="Option__Link"
          href="/test"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-home fa-w-18 Option__Icon"
            data-icon="home"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           
          Test
        </a>
        <a
          aria-current={null}
          className="Option__Link"
          href="/another"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-users fa-w-20 Option__Icon"
            data-icon="users"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           
          Another
        </a>
      </div>
    </div>
    <div
      className="Layout__Content"
    >
      <div
        className="NavBar right"
      >
        <a
          className="NavBar__link menu"
          href="/home/display-menu"
          onClick={[Function]}
        >
          <figure
            className="image"
          >
            <img
              alt="menu"
              className=""
              onError={[Function]}
              src="menu.svg"
            />
          </figure>
        </a>
        <a
          className="NavBar__link"
          href="/"
          onClick={[Function]}
        >
          Cerrar Sesion
        </a>
      </div>
      <div
        className="Layout__Scroll-Content-Wrapper "
      >
        <div
          className="Layout__Scroll-Content"
        >
          Test
        </div>
      </div>
    </div>
  </div>
</div>
`;
